WCCOM and WooJPC Login: make it part of isWhiteLogin

raicem opened the pull request:
Fixes WCCOM-1456

Woo
Part of https://linear.app/a8c/issue/DOTCOM-13397/woo
Part of https://linear.app/a8c/issue/DOTCOM-13385/woo

WooJPC
Part of https://linear.app/a8c/issue/DOTCOM-13625/woojpc
Part of https://linear.app/a8c/issue/DOTCOM-13626/woojpc

## Proposed Changes

**WCCOM**
| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/d5590f08-f571-4e66-8c7a-0083d909c58b) |![image](https://github.com/user-attachments/assets/dfc2c1ca-1d5a-46b5-81ca-1b0023f05f21)| 

**WooJPC (urls in the testing instructions)**
| Before | After |
|--------|--------|
| ![image](https://github.com/user-attachments/assets/05007e49-3d1a-4885-b7cf-094d88c1c159) |![image](https://github.com/user-attachments/assets/6fe63f76-3df3-498a-ba2f-8e24da26d291)| 

## Testing Instructions

- Go to [WooCommerce.com login](http://calypso.localhost:3000/log-in?client_id=50916&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50916%26state%3Dbabd2834d29179c7104bba84668042e1f8a4c310f62299dd24992cf002dbda31%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.com%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%3Dnux%26calypso_env%3Dproduction%26from-calypso%3D1) and confirm it works.
- Mimic going through WooPayments connection: 
    - Users first land on [the signup/authorize page](http://calypso.localhost:3000/jetpack/connect/authorize?client_id=245657653&redirect_uri=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fhandler%3Djetpack-connection-webhooks%26action%3Dauthorize%26_wpnonce%3D9bdd4b194b%26redirect%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-settings%2526tab%253Dcheckout%2526path%253D%252Fwoopayments%252Fonboarding%2526wpcom_connection_return%253D1&state=1&scope=administrator%3Acebfe5edc3805525128c6b79ecbbee7e&user_email=cem.unalan%40a8c.com&user_login=raicem&jp_version=14.8-a.5&secret=pQONgKUOSZeSEJYXRLeQCGAxBpFRyFF1&blogname=Total+Sable+Millipede&site_url=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja&home_url=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja&site_icon&site_lang=en_US&site_created=2025-06-18+15%3A02%3A22&allow_site_connection=1&calypso_env=production&source&_as=wp-cli&locale=en&from=woocommerce-onboarding&plugin_name=woocommerce-payments&color_scheme=fresh&_ui=11140294&_ut=wpcom%3Auser_id&purchase_nonce=aLOfwMCn&_wp_nonce=1c5e04be2f&redirect_after_auth=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-settings%26tab%3Dcheckout%26path%3D%2Fwoopayments%2Fonboarding%26wpcom_connection_return%3D1&site=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja).
    - Fill in your email and see you are prompted to login. Go to the login URL (also find [the URL here](http://calypso.localhost:3000/log-in/jetpack?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fjetpack%2Fconnect%2Fauthorize%3Fclient_id%3D245657653%26redirect_uri%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fhandler%253Djetpack-connection-webhooks%2526action%253Dauthorize%2526_wpnonce%253D9bdd4b194b%2526redirect%253Dhttps%25253A%25252F%25252Ftotal-sable-millipede.jurassic.ninja%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Dwc-settings%252526tab%25253Dcheckout%252526path%25253D%25252Fwoopayments%25252Fonboarding%252526wpcom_connection_return%25253D1%26state%3D1%26scope%3Dadministrator%253Acebfe5edc3805525128c6b79ecbbee7e%26user_email%3Dcem.unalan%2540a8c.com%26user_login%3Draicem%26jp_version%3D14.8-a.5%26secret%3DpQONgKUOSZeSEJYXRLeQCGAxBpFRyFF1%26blogname%3DTotal%2BSable%2BMillipede%26site_url%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%26home_url%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%26site_icon%26site_lang%3Den_US%26site_created%3D2025-06-18%2B15%253A02%253A22%26allow_site_connection%3D1%26calypso_env%3Dproduction%26source%26_as%3Dwp-cli%26locale%3Den%26from%3Dwoocommerce-onboarding%26plugin_name%3Dwoocommerce-payments%26color_scheme%3Dfresh%26_ui%3D11140294%26_ut%3Dwpcom%253Auser_id%26purchase_nonce%3DaLOfwMCn%26_wp_nonce%3D1c5e04be2f%26redirect_after_auth%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-settings%2526tab%253Dcheckout%2526path%253D%252Fwoopayments%252Fonboarding%2526wpcom_connection_return%253D1%26site%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja&from=woocommerce-onboarding&plugin_name=woocommerce-payments)). 
    - Confirm you can log in.
- Mimic going through WooCommerce core profiler:
    - Users first land on [the signup/authorize page](http://calypso.localhost:3000/jetpack/connect/authorize?client_id=245657653&redirect_uri=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fhandler%3Djetpack-connection-webhooks%26action%3Dauthorize%26_wpnonce%3De1fd5da4a3%26redirect%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin&state=1&scope=administrator%3Acebfe5edc3805525128c6b79ecbbee7e&user_email=cem.unalan%40a8c.com&user_login=raicem&jp_version=14.8-a.5&secret=pQONgKUOSZeSEJYXRLeQCGAxBpFRyFF1&blogname=Total%20Sable%20Millipede&site_url=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja&home_url=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja&site_icon=&site_lang=en_US&site_created=2025-06-18%2015%3A02%3A22&allow_site_connection=1&calypso_env=production&source=&_as=wp-cli&locale=en&from=woocommerce-core-profiler&plugin_name=&color_scheme=fresh&_ui=11140294&_ut=wpcom%3Auser_id&purchase_nonce=5ispUNoV&_wp_nonce=1c5e04be2f&redirect_after_auth=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja%2Fwp-admin%2Fadmin.php%3Fpage%3Dwc-admin&site=https%3A%2F%2Ftotal-sable-millipede.jurassic.ninja).
    - Fill in your email and see you are prompted to login. Go to the login URL (also find [the URL here](http://calypso.localhost:3000/log-in/jetpack?redirect_to=http%3A%2F%2Fcalypso.localhost%3A3000%2Fjetpack%2Fconnect%2Fauthorize%3Fclient_id%3D245657653%26redirect_uri%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fhandler%253Djetpack-connection-webhooks%2526action%253Dauthorize%2526_wpnonce%253De1fd5da4a3%2526redirect%253Dhttps%25253A%25252F%25252Ftotal-sable-millipede.jurassic.ninja%25252Fwp-admin%25252Fadmin.php%25253Fpage%25253Dwc-admin%26state%3D1%26scope%3Dadministrator%253Acebfe5edc3805525128c6b79ecbbee7e%26user_email%3Dcem.unalan%2540a8c.com%26user_login%3Draicem%26jp_version%3D14.8-a.5%26secret%3DpQONgKUOSZeSEJYXRLeQCGAxBpFRyFF1%26blogname%3DTotal%2520Sable%2520Millipede%26site_url%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%26home_url%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%26site_icon%3D%26site_lang%3Den_US%26site_created%3D2025-06-18%252015%253A02%253A22%26allow_site_connection%3D1%26calypso_env%3Dproduction%26source%3D%26_as%3Dwp-cli%26locale%3Den%26from%3Dwoocommerce-core-profiler%26plugin_name%3D%26color_scheme%3Dfresh%26_ui%3D11140294%26_ut%3Dwpcom%253Auser_id%26purchase_nonce%3D5ispUNoV%26_wp_nonce%3D1c5e04be2f%26redirect_after_auth%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja%252Fwp-admin%252Fadmin.php%253Fpage%253Dwc-admin%26site%3Dhttps%253A%252F%252Ftotal-sable-millipede.jurassic.ninja&from=woocommerce-core-profiler)).

## Pre-merge Checklist

<!--
Complete applicable items on this checklist **before** merging into trunk. Inapplicable items can be left unchecked.

Both the PR author and reviewer are responsible for ensuring the checklist is completed.
-->

- [ ] Has the general commit checklist been followed? (PCYsg-hS-p2)
- [ ] [Have you written new tests](https://wpcalypso.wordpress.com/devdocs/docs/testing/index.md) for your changes?
- [ ] Have you tested the feature in Simple (P9HQHe-k8-p2), Atomic (P9HQHe-jW-p2), and self-hosted Jetpack sites (PCYsg-g6b-p2)?
- [ ] Have you checked for TypeScript, React or other console errors?
- [ ] Have you used memoizing on expensive computations? More info in [Memoizing with create-selector](https://github.com/Automattic/wp-calypso/blob/trunk/packages/state-utils/src/create-selector/README.md) and [Using memoizing selectors](https://react-redux.js.org/api/hooks#using-memoizing-selectors) and [Our Approach to Data](https://github.com/Automattic/wp-calypso/blob/trunk/docs/our-approach-to-data.md)
- [ ] Have we added the "[Status] String Freeze" label as soon as any new strings were ready for translation (p4TIVU-5Jq-p2)?
    - [ ] For UI changes, have we tested the change in various languages (for example, ES, PT, FR, or DE)? The length of text and words vary significantly between languages. 
- [ ] For changes affecting Jetpack: Have we added the "[Status] Needs Privacy Updates" label if this pull request changes what data or activity we track or use (p4TIVU-aUh-p2)?

raicem committed:
WCCOM Login: make it part of isWhiteLogin

raicem commented on client/layout/logged-out.jsx:
```diff
@@ -161,7 +163,7 @@ const LayoutLoggedOut = ( {
 	}
 
-	if ( isBlazePro && isWhiteLogin ) {
+	if ( ( isBlazePro || isWoo ) && isWhiteLogin ) {
```
Not happy with this, I should find a better way to do this.

chriskmnds replied:
This is the one I stumbled on too, and left it for later (when cleaning up the create-your-account screen)

Just to confirm, this is so the masterbar is imported but not used right (so to get its styles on the page)? Same case with Blaze pro

raicem replied:
Yes, exactly the same reason. I see the masterbar is still used on create an account pages as well. I'm happy to align with whatever solution you might choose in the future

raicem commented on client/layout/logged-out.jsx:
```diff
@@ -146,12 +146,14 @@ const LayoutLoggedOut = ( {
 		'is-blaze-pro': isBlazePro,
 		'two-factor-auth-enabled': twoFactorEnabled,
 		'is-woo-com-oauth': isWooOAuth2Client( oauth2Client ),
+		woo: isWoo,
```
Not happy with this, should find a better solution. Adding `woo` brings woo colors and border radius changes.

chriskmnds replied:
There are also other client color palettes and font sets that need to reintroduced (see https://linear.app/a8c/issue/DOTCOM-13542/login-ensure-custom-typesetsfonts-are-picked-up-for-the-client-screens). I'd need to check how others are overriding component styles in the main controller - ideally we'd use something similar (perhaps it's through the `is-foo` classes). 

For this case, can we remove the border radius holistically, while keeping just the colors? We can do this in another PR if you prefer, but I think that's the idea. cc @jasmussen - do we keep the more rounded borders for the buttons for Woo Login?

jasmussen replied:
It is important for us that we group local customizations together, and minimize them, sure, so that when improvements are made to the overall flow, all log-in consumers inherit those benefits, rather than each login being custom implementations as they are now.

That's our highest goal with this.

We also acknowledge in this that those consumers desire some visual customization options, and honestly we need to still feel out the exact extent of what those options are. As we do feel that out through all these PRs, so long as we keep the code quality value in place, we can experiment a little bit. Just want to avoid crazy hacks, if code gets too complex yet a particular override is needed, it likely needs to become a prop on the componentry so we aviod the hack. This is important so things don't break between updates.

With that said, my instincts for customization so far are: heading font, weight, size. Body font, weight, size. Button color (with WCAG contrast caveat), font weight, size. Button radius. Overall background color, text color. And then some wiggleroom for descriptive text customization.

What should probably *not* be customized: overall layout. Order of componentry (i.e. moving the T&S/Privacy links below create account buttons). Button labels, text phrasing. Hiding labels. Moving links around. 

Does that make sense/help?

chriskmnds replied:
Thanks @jasmussen ! Makes sense, and thanks for clarifying. It was actually not too clear until now how much we can customise in these screens 👍 

> It is important for us that we group local customizations together, and minimize them, sure, so that when improvements are made to the overall flow, all log-in consumers inherit those benefits, rather than each login being custom implementations as they are now.

Indeed, this is a nice target. Once all the clients have migrated to the unified design, there may be opportunities to further clean the code and centralise a little better.

@raicem - so based on the above, we can keep the border radius intact, as expected for Woo. We should bring back the font sets and the color variation where those are missing.

raicem replied:
OK, I have kept the border radius, colors and the font sets. Thank you @chriskmnds and @jasmussen for explanation!

raicem commented on client/blocks/login/login-header.tsx:
```diff
@@ -121,6 +121,7 @@ export function getHeaderText(
 	if ( action === 'lostpassword' ) {
 		headerText = translate( 'Forgot your password?' );
 	} else if ( oauth2Client ) {
+		// TODO: remove isWCCOM when login pages are unified and we have clarified design.
```
I'll remove some of the TODOs after we lock-in the design choices.

chriskmnds commented on client/login/wp-login/index.jsx:
```diff
@@ -517,7 +515,9 @@ export class Login extends Component {
 		const canonicalUrl = localizeUrl( 'https://wordpress.com/log-in', locale );
-		const isSocialFirst = isWhiteLogin && ! isGravPoweredClient && ! isWoo;
+
+		// TODO: remove isGravPoweredClient when login pages are unified.
```
Brings a smile on my face that `isSocialFirst === isWhiteLogin` within the next couple of PRs. Both on the way out 🧹

chriskmnds reviewed (commented):
Thank you so much for owning this! ❤️ 

A couple of observations (more like confirmations) from testing with the Live image:

- I believe the font needs to stay as in the original, which for the heading is `--woo-font-family-header: proxima-vara, "Proxima Fallback", sans-serif;`. Right now, it reverted to Recoleta, which is the same case with other OAuth client screens we migrated (mistakenly so, see https://linear.app/a8c/issue/DOTCOM-13542/login-ensure-custom-typesetsfonts-are-picked-up-for-the-client-screens). Can we bring back the Woo font?

- The TOS at the bottom need to be removed. I believe [this](https://github.com/Automattic/wp-calypso/pull/103948/files#diff-a31cf50bd8c0e5b1e90b3683eac0fe8e0d44c0a90c0b74d9042f7291d05fc036R329) is the condition we need to update.

  > By continuing with any of the options above, you agree to our [Terms of Service](https://wordpress.com/tos/) and have read our [Privacy Policy](https://automattic.com/privacy/).

- `Log in to your account` should actually be `Log in to Woo with WordPress.com`. It's the parts in `getHeaderText` (I think you have a comment/TODO in the code  for this, right?)

- Shall we remove the custom border radius?

<img width="1125" alt="Screenshot 2025-06-10 at 9 04 42 PM" src="https://github.com/user-attachments/assets/b73522c2-ce98-4b82-9830-821800d49064" />

raicem committed:
Woo Login: apply Woo styles

raicem committed:
Woo Login: remove unneeded social buttons styling

raicem committed:
Woo Login: address PR feedback

raicem committed:
Woo Login: add WooOauthMasterbar

It's also used on the create account page, and removing it breaks the
styling there.

raicem committed:
Woo Login: remove a TODO

raicem reviewed (commented):
Thank you @chriskmnds for the review! I have pushed up all of my changes.

> Can we bring back the Woo font?

:heavy_check_mark: Done, I realized one of later commits last night broke things badly, but it's back now. 

![image](https://github.com/user-attachments/assets/b225bc68-e589-4b5f-9fdf-fa46d23e24c4)

> The TOS at the bottom need to be removed
> Log in to your account should actually be Log in to Woo with WordPress.com

:heavy_check_mark: All done and pushed up to the remote.

> Shall we remove the custom border radius?

I left it in there as per the other [discussion](https://github.com/Automattic/wp-calypso/pull/104104/files#r2139411536) thread. I have [removed](https://github.com/Automattic/wp-calypso/pull/104104/commits/025b4881d37d48444ee3c941e644b0622949956b) a lot of CSS about the social login elements though. I believe there is still more to remove, when moving to other pages.


Apart from this work, I think WooJPC flows need a bit more love. I would like to enable the logo and a bit more styling there. Maybe that can be another PR?

I'm also bit worried that we lost the subheading information [that we have on production](https://wordpress.com/log-in?from=woocommerce-core-profiler). That being said, I don't want to further tangle things in this PR.

![image](https://github.com/user-attachments/assets/f3694236-bde3-425f-852e-19063652cac4)

raicem committed:
Woo Login: remove unneeded import

chriskmnds commented on client/blocks/login/login-header.tsx:
```diff
@@ -137,7 +135,7 @@ export function getHeaderText(
 				);
 			}
-			headerText = translate( 'Log in to your account' );
+			headerText = translate( 'Log in to Woo with WordPress.com' );
```
This should actually be handled already up on top inside the `isSocialFirst`: https://github.com/Automattic/wp-calypso/pull/104104/files#diff-fc758b906c257441858a1ebb10f7aa7360fb06563cd3962d011770a777cd2a61R75

Isn't that one picked up already? So no need to update here - other than remove completely? 🤔

raicem replied:
Sorry, forgot to reply to this. This is done now.

chriskmnds commented on client/layout/masterbar/woo.scss:
```diff
@@ -445,45 +443,6 @@ $breakpoint-mobile: 660px;
 		margin-top: 25px;
 	}
 
-	.auth-form__social-buttons {
```
Nice! Removing obsolete styles is the right way!

chriskmnds commented on client/layout/masterbar/woo.scss:
```diff
@@ -1272,7 +1168,8 @@ $breakpoint-mobile: 660px;
 		h2,
-		h3 {
+		h3,
+		.wp-login__heading-text {
```
We need to think where these are best to be defined more centrally. Probably not in the masterbar. 
I haven't got around to studying this, but thinking the controller overrides, or layout main style, or something there (so we also move whatever remains from masterbar styles - from woo and blaze)

chriskmnds commented on client/login/wp-login/components/heading-logo.tsx:
```diff
@@ -32,6 +35,17 @@ const HeadingLogo = ( { isFromAkismet }: Props ) => {
 		logo = <img src={ blazeProLogo } alt="Blaze Pro Logo" />;
+	} else if ( isWooOAuth2Client( oauth2Client ) ) {
+		logo = (
+			<SVGIcon
+				name="woocommerce-logo"
+				icon={ WooLogo }
+				classes="masterbar__woo-client-logo"
+				width="128"
+				height="48"
+				viewBox="0 0 64 24"
+			/>
```
Also subject to cleanup along with masterbar (the "masterbar" class here)

chriskmnds reviewed (commented):
You are a star, thank you @raicem ! 🌟 
Code looks great. Left some comments, but I think nothing blocking - maybe except the clarity around the [header text](https://github.com/Automattic/wp-calypso/pull/104104/files#r2140486702) (where it comes from).

> I'm also bit worried that we lost the subheading information [that we have on production](https://wordpress.com/log-in?from=woocommerce-core-profiler). That being said, I don't want to further tangle things in this PR.

Indeed, it does feel like there was an intention with it. Maybe we attempt to reintroduce it in a follow-up PR (but be open to abandoning it) :-)

 > Apart from this work, I think WooJPC flows need a bit more love. I would like to enable the logo and a bit more styling there. Maybe that can be another PR?

Yes, another PR sounds good. Happy to review any follow-ups, so pls keep them coming :-) 

Happy to test this tomorrow and review anything else you'd like. But also go for it if you feel confident or get another review. There are some conflicts that I probably introduced earlier

chriskmnds reviewed (commented):
Thanks again @raicem 
I did some testing. The Woo Login looks great. 

The WooJPC we can tune a little, as also the lost-password screen looks broken right now. I think perhaps we need to:
- ensure we don't render the masterbar in lost-password form (for WooJPC)
- add the Woo logo (for WooJPC)
- adjust the width of the main container for the inputs/buttons in main Login screen (for WooJPC)

Any heavier lifts we can address separately. Some screens with highlights below:

| lost-password | main Login |
|--------|--------|
| <img width="929" alt="Screenshot 2025-06-17 at 12 02 06 PM" src="https://github.com/user-attachments/assets/16ecdb62-3222-4b1a-8b42-0de42bde017b" /> | <img width="1027" alt="Screenshot 2025-06-17 at 12 03 00 PM" src="https://github.com/user-attachments/assets/12d3a0bf-b163-4ec9-851c-00160cc888fd" /> | 

p.s. thanks for pointing at the WooJPC screens. I was not aware these also existed. I've added respective issues in Linear and linked them to the PR

raicem committed:
Woo Login: move headerText logic under isSocialFirst block

raicem commented:
Thank you again for checking @chriskmnds! I'm back from AFK and gettting back into it. I'll work on WooJPC screens now and align them better with what we have in this PR.

I also noticed we influenced continue as user pages (both for [Woo](http://calypso.localhost:3000/log-in?client_id=50019&redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%2F%3Fresponse_type%3Dcode%26client_id%3D50019%26state%3D0e24650106e21ba73b419e4ba56623cf4c0f139ad3ea0249361b78d0a72fa724%26redirect_uri%3Dhttps%253A%252F%252Fwoocommerce.test%252Fwc-api%252Fwpcom-signin%253Fnext%253D%25252F%26blog_id%3D0%26wpcom_connect%3D1%26wccom-from%26calypso_env%3Dproduction%26from-calypso%3D1) and [Blaze](https://wordpress.com/log-in?redirect_to=https%3A%2F%2Fpublic-api.wordpress.com%2Foauth2%2Fauthorize%3Fresponse_type%3Dcode%26client_id%3D99370%26state%3D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb190dc03654cef0465786da0%26redirect_uri%3Dhttps%253A%252F%252Fblazepro.tumblr.com%252Fauth%252Fconnected%26blog_id%3D0%26wpcom_connect%3D1%26calypso_env%3Dproduction%26scope%3Dglobal%26from-calypso%3D1&client_id=99370&signup_url=%2Fstart%2Fwpcc%2Foauth2-user%3Foauth2_client_id%3D99370%26oauth2_redirect%3Dhttps%253A%252F%252Fpublic-api.wordpress.com%252Foauth2%252Fauthorize%253Fresponse_type%253Dcode%2526client_id%253D99370%2526state%253D0f0d1f16975b65cf8e9e924461b7c892f78eabeeb190dc03654cef0465786da0%2526redirect_uri%253Dhttps%25253A%25252F%25252Fblazepro.tumblr.com%25252Fauth%25252Fconnected%2526blog_id%253D0%2526wpcom_connect%253D1%2526calypso_env%253Dproduction%2526scope%253Dglobal%2526from-calypso%253D1)). They ended up with a double ToS. [This commit](https://github.com/Automattic/wp-calypso/pull/104104/commits/1fda7f90f85cfe2d12420fbfe2f5f523db116cc3) fixes it but I'm happy to pull that out in a new PR to ship it faster.

raicem committed:
WooJPC Login: update styling

raicem committed:
WooJPC Login: remote Jetpack styling and align with Woo styling

raicem commented:
OK, WooJPC changes are done, I have documented how they are accessed by users [in the Linear issue](https://linear.app/a8c/issue/DOTCOM-13625/woojpc), also updated testing instructions and screenshots here. Also rebased on top of trunk.

@chriskmnds sorry about the back and forth on this PR but I made some further changes. For the latest point you checked, to today's iteration, [here is the list of commits and the diff](https://github.com/Automattic/wp-calypso/compare/af309fcd5f4670549cba1ca90c205ea9f748ac23...21fd7e1011e6d9f465b2bf9a5b9203932d655168), in case you want to review.

raicem committed:
Woo Login: remove unused wccomFrom prop

raicem committed:
Login: don't show ToS on lost password page

raicem committed:
Woo Login: remove lost password page customizations

raicem committed:
Login: don't render ToS in HeadingSubText when on lost password page

chriskmnds commented on client/blocks/login/login-form.jsx:
```diff
@@ -1080,7 +1080,13 @@ export class LoginForm extends Component {
 			! isGravatarFlowWithEmail;
 
-		if ( isJetpack ) {
+		/**
+		 * This is the Jetpack login form. Some Woo logins (WooPayments, Core Profiler) also part of
+		 * Jetpack login but they must have consistent styling across all Woo logins.
+		 *
+		 * TODO: remove this once we have consistent styling for Jetpack as well.
+		 */
```
Thanks for the clarification!

chriskmnds commented on client/blocks/login/login-header.tsx:
```diff
@@ -79,6 +78,16 @@ export function getHeaderText(
+		} else if ( isWCCOM ) {
+			if ( twoFactorEnabled ) {
+				headerText = translate( 'Authenticate your login' );
+			} else if ( currentQuery.lostpassword_flow ) {
+				headerText = translate( 'Log in to your account' );
+			} else if ( showContinueAsUser ) {
+				headerText = translate( 'Log in to your account' );
+			}
```
Just curious why we need to keep these 🤔

raicem replied:
I thought they do something, but they don't work. So removed them now.

chriskmnds commented on client/blocks/login/login-header.tsx:
```diff
@@ -79,6 +78,16 @@ export function getHeaderText(
+				headerText = translate( 'Log in to your account' );
+			}
+
+			headerText = translate( 'Log in to Woo with WordPress.com' );
```
I think for the default case, it might be better to set the `clientName` here as `Woo` and not set the `headerText`. But NIT - I'm sure this will evolve more as we continue with the unification work.

chriskmnds commented on client/layout/logged-out.jsx:
```diff
@@ -147,12 +147,14 @@ const LayoutLoggedOut = ( {
 	let masterbar = null;
 
 	useEffect( () => {
+		// TODO: remove this useEffect when login pages are unified.
```
Thanks! What does this (`refreshColorScheme`) do exactly / why is it needed?

raicem replied:
I see it adds a `is-fresh` class to the body, for WooJPC pages. There is a [mechanism](https://github.com/Automattic/wp-calypso/blob/3a1c6c1888ebcd0822d1d03874967f4fc9760c47/client/layout/index.jsx#L404) to apply a theme based on the URL parameter, that's specifically used for WooJPC. I just found this out.

I'm not sure if that's still in use, might be hard to discover how it's used. So I kept that in place. Updated the TODO.

![image](https://github.com/user-attachments/assets/ead3491c-3326-491e-8cf5-96c2478fd7de)

chriskmnds replied:
Thanks for the notes! I'll take a closer look too, later on.

chriskmnds commented on client/layout/logged-out.jsx:
```diff
@@ -377,6 +380,7 @@ export default withCurrentRoute(
 				isInStepContainerV2FlowContext( currentRoute, currentQuery );
 			const twoFactorEnabled = isTwoFactorEnabled( state );
 
+			// TODO: remove isWooJPC check.
```
Can you add a few more notes on what this is used for (in the TODO/code)? 🙇‍♂️

raicem replied:
Sure thing, done now

chriskmnds commented on client/login/wp-login/style.scss:
```diff
@@ -83,7 +83,7 @@ $image-height: 47px;
 }
 
-.wp-login__main.main {
+.layout:not(.woo) .wp-login__main.main {
```
I think this could be `,is-white-login` instead of `.woo`, but not sure (hopefully it is). I think other screens may suffer the same. Can we add the comment below ("clk" is just my initials):

```suggestion
/* TODO clk consider .is-white-login instead of .woo */
.layout:not(.woo) .wp-login__main.main {
```

raicem replied:
I think you (or someone else) already removed this block of code in another PR, rebasing it got rid of it fully.

chriskmnds reviewed (approved):
Amazing! This looks great. Thanks @raicem I left some comments (mostly questions for me to understand). 

The only bit I noticed is a slightly different style for the "lost your password" link in the WooJPC flows (this should match the "create an account" link above, or what's in the Woo Login screen - bigger font, underlined):

| Woo | WooJPC |
| --- | --- |
| <img width="1089" alt="Screenshot 2025-06-19 at 6 30 11 PM" src="https://github.com/user-attachments/assets/734cfe3a-8a22-499b-8f0e-b395f2487759" /> | <img width="1024" alt="Screenshot 2025-06-19 at 6 27 37 PM" src="https://github.com/user-attachments/assets/82897d77-6b2f-4ad7-96f4-d7a1ac02ca9e" /> |

Also, not sure if related, but I cannot access `/log-in/jetpack` from the Calypso live image. It throws an error in the console, but not sure if related. It does work fine in production. Perhaps check another PR if the same and it's just a WPCalypso/env issue. Please confirm this before merging.

For testing more things, I would normally try the default WP login and a couple of other OAuth clients briefly from [the unified list](https://linear.app/a8c/issue/DOTCOM-13220/login-make-the-two-columnwhite-layout-the-default-across-all-contexts) (to make sure nothing broke). Gravatar Login won't be unified, so worth just quick test to make sure nothing weird there.

raicem committed:
Woo Login: fix Jetpack logo appearing instead

raicem committed:
Woo Login: remove unused code

raicem committed:
Woo and Blaze login: prevent footer ToS in Continue as User page

raicem committed:
Woo Login: remove unused header elements

raicem committed:
WooJPC: update a Todo

raicem commented:
Thank you again @chriskmnds! 

> Also, not sure if related, but I cannot access /log-in/jetpack from the Calypso live image. It throws an error in the console, but not sure if related

I can access that URL now with no issues. In fact, I went through a lot of flows and I think this is ready to go. It's Friday afternoon for me, so I'm inclined to do it on a Monday morning.

raicem committed:
WooJPC Login: remove tests checking WooPayments sub header text

Since logins are now unified with task DOTCOM-13385, WooPayments no
longer has this special subheader text. This commit removes the tests
regarding that copy.

raicem committed:
Login: remove invalid parameter from getHeaderText function

chriskmnds commented on client/login/wp-login/index.jsx:
```diff
@@ -576,13 +571,17 @@ export class Login extends Component {
 		}
 
+		const isLostPassword =
+			currentRoute === '/log-in/lostpassword' || currentRoute === '/log-in/jetpack/lostpassword';
```
I'm also working on this in parallel for https://github.com/Automattic/wp-calypso/pull/104367, but no worries, this PR should be merged first.

I think we can send the `action` prop instead to `HeadingSubText`, which will have the value `lostpassword` for the lost-password screen. Can you try it with that instead? :-)

raicem replied:
Thanks @chriskmnds for checking again!

> I think we can send the action prop instead to HeadingSubText, which will have the value lostpassword for the lost-password screen. Can you try it with that instead? :-)

Nice, that would be way better than this check (I added it last minute, I noticed password reset pages had the ToS).

I just tried the `action` prop, but it was `null` in every case. Before diving in deeper, and delay this PR any further I think I'll go for a merge now. Happy to address this in a subsequent PRs.

chriskmnds replied:
Interesting. It worked for me, but I wasn't testing the Woo flow. 🤔 Glad that we merged this.

The `action` comes from some login context and was used in other places including for Woo login e.g. in footer. I'll dig around.

Thanks for seeing this PR to completion! 🎉
